### PR TITLE
Flight ready

### PR DESCRIPTION
**Pull request by jacoffey3**

### CODE REVIEW

**Comment by jacoffey3** [comment]
@CatGarab @plauche can I get some eyes on the broad strokes? I think I'm leaving a lot out, but I just want some assistance with the outline.

**Comment by jacoffey3** [comment]
[Mission Dev: Flight Readiness doc](https://trello.com/c/YsjL3KBr/881-mission-dev-flight-readiness-doc)

File: docs/mission-dev/flight-ready.rst [hunk 1]
```diff
@@ -1,4 +1,35 @@
...
+This is *only* valid for the software, although it might reference hardware to give context.
+Also, it is important to add steps to account for your mission specific apps and services (or whatever else you add to the system).
+
+Prepare Linux
+-------------
+
+1. Build the flight linux image, including any core and/or hardware services you need for your mission.
+2. Load the flight image onto the spacecraft.
+3. Build the recovery and upgrade rootfs images.
+4. Load them into the recovery and upgrade partitions.
```
Comment (CatGarab) [inline docs/mission-dev/flight-ready.rst]: Recovery/upgrade images are the same thing, not two separate entities (similarly, there's only the one upgrade partition)
The base recovery image is automatically created and installed in the upgrade partition as part of the KLB build process

File: docs/mission-dev/flight-ready.rst [hunk 2]
```diff
@@ -1,4 +1,35 @@
...
+
+Prepare Services
+----------------
+
+1. Boot up the bus.
+2. Ensure any included hardware services are properly communicating with hardware by issuing a ``noop`` mutation to each.
+3. Ensure any payload services you're including are properly communicating with their respective payload (hopefully by running the ``noop`` mutation you dutifully included).
+4. Ensure your ``config.toml`` is appropriate for your bus configuration.
+5. Configure your :doc:`logging and log retention <../os-docs/linux-docs/logging>` for your system memory requirements.
+  a. We recommend that you limit logging to ``info`` and higher if you are using a file location that is prone to wearing out, as ``debug`` logs can cause significant amounts of logs to be generated.
```
Comment (CatGarab) [inline docs/mission-dev/flight-ready.rst]: Good point. Probably add a link to the app framework docs pointing out where the log level is set ([python](https://docs.kubos.com/1.15.0/app-docs/python-app-api.html#app_api.logging_setup), [rust](https://docs.kubos.com/1.15.0/rust-docs/kubos_app/fn.app_start.html))

File: docs/mission-dev/flight-ready.rst [hunk 3]
```diff
@@ -1,4 +1,35 @@
...
+
+Prepare Applications
+--------------------
+
+1. Register stable versions of all applications (if you include multiple versions, make sure the correct one is active)
+2. Reboot the bus, and ensure all apps that should start on boot are properly started.
+3. If you followed the :doc:`Deployment doc <deployment>` for your deployment app, or you are using the KubOS deployment app, you will need to:
+  a. Ensure the ``deployed`` U-boot environment variable is set to ``False``.
+  b. Ensure the ``remove_before_flight`` U-boot environment variable is set to ``False``.
+  c. Ensure the ``deploy_start`` U-boot environment variable is unset.
```
Comment (CatGarab) [inline docs/mission-dev/flight-ready.rst]: Mention the `fw_setenv` command, which is how users can set uboot envars from the linux command line

File: docs/mission-dev/flight-ready.rst (hunk 1 above)
Comment (jacoffey3) [reply docs/mission-dev/flight-ready.rst]: What step should be listed here then? Verify it?

File: docs/mission-dev/flight-ready.rst (hunk 2 above)
Comment (jacoffey3) [reply docs/mission-dev/flight-ready.rst]: Can we not just set it for system-wide?

File: docs/mission-dev/flight-ready.rst (hunk 3 above)
Comment (jacoffey3) [reply docs/mission-dev/flight-ready.rst]: Basically this? 
Note: the ``fw_setenv`` command is useful for viewing/setting these uboot envars.

File: docs/mission-dev/flight-ready.rst (hunk 2 above)
Comment (CatGarab) [reply docs/mission-dev/flight-ready.rst]: Oh right. In that case, the rsyslog configuration files would need to be updated.
Remove the two lines containing "debug" out of the app and/or service config files:
- [app config](https://github.com/kubos/kubos-linux-build/blob/master/common/overlay/etc/rsyslog.d/kubos-apps.conf)
- [service config](https://github.com/kubos/kubos-linux-build/blob/master/common/overlay/etc/rsyslog.d/kubos-services.conf)

File: docs/mission-dev/flight-ready.rst (hunk 3 above)
Comment (CatGarab) [reply docs/mission-dev/flight-ready.rst]: How about this:

Note: The [fw_setenv](http://manpages.ubuntu.com/manpages/xenial/man8/fw_setenv.8.html) command can be used to set U-Boot environment variables from the Kubos Linux command line. Similarly, the [fw_printenv](http://manpages.ubuntu.com/manpages/xenial/man8/fw_printenv.8.html) command can be used to print the current value of a U-Boot environment variable.

File: docs/mission-dev/flight-ready.rst (hunk 1 above)
Comment (CatGarab) [reply docs/mission-dev/flight-ready.rst]: Yeah. `3. Verify that the recovery and upgrade processes work as expected`

(recovery process is triggered by either manually corrupting the system or setting `boot_count` to 2. Upgrade process is triggered by setting `kubos_updatefile` with the upgrade file name)

Comment (jacoffey3) [reply docs/mission-dev/flight-ready.rst]: Should I recommend they go through that proceed to prepare for flight?

File: docs/mission-dev/flight-ready.rst [hunk 4]
```diff
@@ -1,4 +1,41 @@
...
-All the things you need to do software-wise to finalize your mission
\ No newline at end of file
+This checklist is a **framework** for when you are preparing KubOS to integrate your spacecraft into the launch vehicle.
+This is *only* valid for the software, although it might reference hardware to give context.
+Also, it is important to add steps to account for your mission specific apps and services (or whatever else you add to the system).
+
+Prepare Linux
+-------------
+
+1. Build the flight linux image, including any core and/or hardware services you need for your mission.
```
Comment (CatGarab) [inline docs/mission-dev/flight-ready.rst]: nit: capitalize Linux. Should atleast be Kubos Linux. Should it maybe be KubOS instead?

Comment (jacoffey3) [inline docs/mission-dev/flight-ready.rst]:
```suggestion
1. Build the flight KubOS image, including any core and/or hardware services you need for your mission.
```

**Pull request merged**